File: src/main/java/com/vjiki/music/controller/SongLikeController.java
package com.vjiki.music.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.vjiki.music.dto.SongLikeRequest;
import com.vjiki.music.service.SongLikeService;

@RestController
@RequestMapping("/api/v1/song-likes")
public class SongLikeController {

    private final SongLikeService songLikeService;

    public SongLikeController(SongLikeService songLikeService) {
        this.songLikeService = songLikeService;
    }

    @PostMapping("/like")
    public ResponseEntity<Void> likeSong(@RequestBody SongLikeRequest request) {
        songLikeService.likeSong(request.getUserId(), request.getSongId());
        return ResponseEntity.ok().build();
    }

    @PostMapping("/dislike")
    public ResponseEntity<Void> dislikeSong(@RequestBody SongLikeRequest request) {
        songLikeService.dislikeSong(request.getUserId(), request.getSongId());
        return ResponseEntity.ok().build();
    }
}

File: src/main/kotlin/com/vjiki/music/controller/SongLikeController.kt
package com.vjiki.music.controller

import com.vjiki.music.dto.SongLikeRequest
import com.vjiki.music.service.SongLikeService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.UUID

@RestController
@RequestMapping("/api/v1/song-likes")
class SongLikeController(
    private val songLikeService: SongLikeService
) {

    @PostMapping("/like")
    fun likeSong(@RequestBody request: SongLikeRequest): ResponseEntity<Void> {
        songLikeService.likeSong(request.userId, request.songId)
        return ResponseEntity.ok().build()
    }

    @PostMapping("/dislike")
    fun dislikeSong(@RequestBody request: SongLikeRequest): ResponseEntity<Void> {
        songLikeService.dislikeSong(request.userId, request.songId)
        return ResponseEntity.ok().build()
    }
}

